annonatate/flaskserver.py

```diff
@@ -154,70 +158,29 @@ def removecollaborator():
-        iiifpath = os.path.join(app.config['UPLOAD_FOLDER'], iiiffolder)
-        infocontents = open(os.path.join(iiifpath, 'info.json')).read()
-        imgurl = "https://%s/"%(prefix)
-        url = "{}{}".format(imgurl, iiiffolder)
-        createjekyllfile(infocontents,'info.json', iiifpath)
-    if iiifimage != 'uploadimage':
-        manifesturl = "{}{}/manifest.json".format(session['origin_url'], manifestpath)
-        manifest = createmanifest(tmpfilepath, imgurl, url, iiiffolder, request.form, manifesturl)
-        if type(manifest) == dict:
-            return render_template('upload.html', error=manifest['error'])
-        else:
-            manifest = manifest.toString(compact=False).replace('canvas/info.json', 'info.json').replace('https://{{site.url}}', '{{site.url}}')
-    if iiifimage == 'uploadimage':
-        file =request.files['file']
-        encodedimage = file.stream.read()
-        response = sendgithubrequest(file.filename, encodedimage, "images").json()
-        uploadtype='image'
+    image = anno.Image(request.form, request.files, session['origin_url'])
+
+    if not image.isimage:
+        response = sendgithubrequest("manifest.json", image.manifest_markdown, image.manifestpath).json()
+        uploadtype = 'manifest'
         if 'content' in response.keys():
-            uploadurl = "{}{}".format(session['origin_url'], response['content']['path'])
-            output =  True
+            uploadurl ='{}{}'.format(image.origin_url, response['content']['path'].replace('_manifest', 'manifest'))
+            output = True
         else:
             output = response['message']
-    elif 'http' in iiifimage:
-        manifest = "---\n---\n{}".format(manifest)
-        response = sendgithubrequest("manifest.json", manifest, manifestpath).json()
-        uploadtype = 'manifest'
+        if type(image.manifest) == dict:
+            return render_template('upload.html', error=image.manifest['error'])
+    else:
+        response = sendgithubrequest(image.file.filename, image.encodedimage, "images").json()
+        uploadtype='image'
         if 'content' in response.keys():
-            uploadurl ='{}{}'.format(session['origin_url'], response['content']['path'].replace('_manifest', 'manifest'))
-            output = True
+            uploadurl = "{}{}".format(image.origin_url, response['content']['path'])
+            output =  True
         else:
             output = response['message']
-    elif iiifimage == 'uploadiiif':
-        createjekyllfile(manifest, 'manifest.json', iiifpath)
-        os.remove(tmpfilepath)
-        output = os.path.join(app.config['UPLOAD_FOLDER'], iiiffolder)
-        shutil.make_archive(output, 'zip', iiifpath)
-        shutil.rmtree(iiifpath)
-        output = "{}.zip".format(iiiffolder)
     triggerbuild()
+    template_name = 'uploadsuccess.html'
+    template_vars = {'output': output, 'uploadurl': uploadurl, 'uploadtype': uploadtype}
```

Comment:
Not sure what these two lines of code are there for?

Reply:
Removed.